pixel-saver@deadalnix.me/decoration.js

```diff
@@ -205,10 +207,14 @@ function onWindowAdded(ws, win) {
 	 * (see workspace.js _doAddWindow)
 	 */
 	if (!win.get_compositor_private()) {
-		Mainloop.idle_add(function () {
-			onWindowAdded(ws, win);
+		if (--compositor_retries) {
```

Comment:
I would go for

compositor_retries = compositor_retries ? compositor_retries - 1 : 2;